components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/openjdk/nashorn/JsOpenJdkNashornGraphBuilder.java

```diff
@@ -1221,7 +1223,18 @@ public Object evaluate(AuthenticationContext authenticationContext, Object... pa
                     }
 
                 } catch (Throwable e) {
-                    //We need to catch all the javascript errors here, then log and handle.
+                    // We need to catch all the javascript errors here, then log and handle.
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog
+                                .DiagnosticLogBuilder(FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK,
+                                "execute-adaptive-script");
+                        diagnosticLogBuilder.resultMessage("Error in executing the adaptive authentication script : " +
+                                        e.getMessage())
+                                .inputParam("application name", authenticationContext.getServiceProviderName())
+                                .resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                                .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION);
+                        LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+                    }
```

Comment:
Can we refactor this to a method. The same thing is used in `JsNashornGraphBuilder.java` as well